docs/guides/calls.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "Seamless Sequential and Parallel Calls in Tailcall"
+---
+
+Building data access layers often involves meticulous orchestration of API calls, but Tailcall simplifies this process. By analyzing your defined schema, it automatically determines the optimal execution strategy, deciding when to sequence calls and when to run them in parallel. This allows you to focus on your core application logic, while Tailcall handles the optimization seamlessly. Now, let's get into some real-world examples to illustrate its functionality.
+
+## Examples
+
+### Example 1: Fetching Comments and Users
+
+Imagine you're building a social media platform and want to display a post with its associated comments and their respective authors.
+
+**Schema:**
+
+```graphql
+type Query {
+  posts: [Post] @http(path: "/posts")
+}
+
+type Post {
+  id: Int!
+  userId: Int!
+  title: String!
+  body: String!
+  user: User
+    @http(path: "/users", query: [{key: "id", value: "{{value.userId}}"}], matchPath: ["id"], matchKey: "userId")
+}
```

Comment:
We don't have `matchPath` and `matchKey` anymore. I think in this example we should talk about the n + 1 usecase. Where for each post we are querying it's user. We should emphasize on the fact that tailcall automatically identifies that all the requests to `/users/{{value.userId}}` should go in parallel.

Reply:
This could be an example of making parallel calls.

Reply:
Sounds good.
I moved this example to be the third to keep the first 2 examples more straightforward